### PR TITLE
fix bug when running `google/paligemma-3b-mix-224`

### PR DESCRIPTION
**Pull request by kaixuanliu**
Results are error when running this test code:
```python
from transformers import AutoProcessor, PaliGemmaForConditionalGeneration
from PIL import Image
import requests
import torch
from habana_frameworks.torch.hpu import wrap_in_hpu_graph
from optimum.habana.transformers.modeling_utils import adapt_transformers_to_gaudi

model_id = "google/paligemma-3b-mix-224"

url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/tasks/car.jpg?download=true"
image = Image.open(requests.get(url, stream=True).raw)
adapt_transformers_to_gaudi()
model = PaliGemmaForConditionalGeneration.from_pretrained(model_id)
device = torch.device("hpu")
model = model.to(device)
model.eval()

processor = AutoProcessor.from_pretrained(model_id)
prompt = "caption es"
model_inputs = processor(text=prompt, images=image, return_tensors="pt")
model_inputs = model_inputs.to(device)
input_len = model_inputs["input_ids"].shape[-1]

with torch.inference_mode():
    generation = model.generate(**model_inputs, max_new_tokens=100, do_sample=False)
    generation = generation[0][input_len:]
    decoded = processor.decode(generation, skip_special_tokens=True)
    print(decoded)
```

This PR help solve this bug.

### CODE REVIEW

**Comment by kaixuanliu** [comment]
Original implementation used `_gaudi_prepare_4d_causal_attention_mask` to replace `self._update_causal_mask`, whereas when the  `len(attenstion_mask.shape)==4`, it will replace `value 0` in attention mask with `torch.finfo(torch.float32).min`, [L95~L98](https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/transformers/modeling_attn_mask_utils.py#L95~L98), and when calculating `attn_weights`, there is an add op [L100](https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/transformers/models/gemma/modeling_gemma.py#L100), which causes the result is wrong.

**Comment by kaixuanliu** [comment]
@regisss ,@lkk12014402 , pls help review

**Comment by regisss** [comment]
When I run the test code on main or on your fork, I get an empty decoded generation. Is that expected?

**Comment by kaixuanliu** [comment]
> When I run the test code on main or on your fork, I get an empty decoded generation. Is that expected?

This is the bug. Expected output should be: "Un auto azul estacionado frente a un edificio". You can refer to this: [HF model](https://huggingface.co/google/paligemma-3b-mix-448)

**Comment by kaixuanliu** [comment]
@regisss @libinta , Can you help re-validate the output of my fork? I checked again and on my fork the output is not empty, it's "Un auto azul estacionado frente a un edificio"

**Comment by kaixuanliu** [comment]
@yao-matrix, here FYI

**Review by regisss — APPROVED** [review]
